### PR TITLE
Refactors vibebots into basic mobs.

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

I don't like writing AI so I don't think I'll be able to do too many more bots, but here's the vibebot. Basically all it does is that it just follows people around while blinking random colors, kinda like the old version but just a bit more enhanced. Does all the same things you might expect from a normal bot though.

Did some code cleanup in the area as well, the behavior should all be a bit more abstracted with plenty of hooks to work into the logic.
## Why It's Good For The Game

Smarter and cooler vibebots to brighten up your station. Also some nice new code improvements to cut out the middleman in a lot of instances.
## Changelog
:cl:
refactor: Vibebots have been refactored to follow people around (at times) and their constant light-spam should be a tad bit more intelligent, let us know if you see any unexpected behavior.
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
> maybe u should make the vibe behavior delete it's target after it's done vibing and add them to the BB_TEMPORARY_IGNORE list so that the bot can search for new people to vibe with. maybe we should create a new targeting datum for bots that skips targets in said list. also might be worthwhile to check if the target is reachable so the bot doesn't get stuck trying to reach someone it can't. I already created a proc that checks whether they're reachable and caches them into the temporary ignore list if they aren't in the basic bot pr

lol github never notified me about this reply or that the fact that this pr was commented

a lot of the nuances of botcode went over my head making this pr so idek what you're saying

File: code/modules/mob/living/basic/bots/vibebot/vibebot_action.dm [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	button_icon_state = "funk"
+
+/datum/action/innate/vibe/Grant(mob/grant_to)
+	. = ..()
+	RegisterSignal(grant_to, COMSIG_BOT_TURNED_ON, PROC_REF(activate_lights))
+	RegisterSignal(grant_to, COMSIG_CHANGE_VIBEBOT_COLOR, PROC_REF(change_color))
+	RegisterSignal(grant_to, COMSIG_BOT_TURNED_OFF, PROC_REF(turn_off_lights))
+
+/datum/action/innate/vibe/Remove(mob/removed_from)
+	UnregisterSignal(removed_from, list(COMSIG_BOT_TURNED_ON, COMSIG_BOT_TURNED_OFF))
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/basic/bots/vibebot/vibebot_action.dm:19]: You're not unregistering ``COMSIG_CHANGE_VIBEBOT_COLOR``?

Comment (san7890) [reply code/modules/mob/living/basic/bots/vibebot/vibebot_action.dm:19]: whoops i forgot

**Comment by san7890** [comment]
ehhh in retrospect i'm not terribly happy with the code that i've written here, anyone else is free to salvage the scraps but i don't think it's fair to leave the tracker clogged up with something i'm not very proud of anymore

**Pull request closed without merging**